src/client/renderer/r_lighting.c

```diff
@@ -386,12 +386,26 @@ static void R_CastShadows(r_lighting_t *l, const r_illumination_t *il) {
 
 		s->illumination = il;
 		s->plane = &r_model_state.world->bsp->planes[tr.plane.index];
-		s->plane->flags |= R_PLANE_SHADOW;
+		s->plane->num_shadows++;
 		s->shadow = shadow;
 	}
 }
 
 /**
+ * @brief Frees the specified shadow, decrementing its plane's shadow count.
+ */
+static void R_FreeShadow(r_shadow_t *s) {
+
+	if (s->plane) {
+		if (s->plane->num_shadows) {
+			s->plane->num_shadows--;
```

Comment:
And here is where she'll crash now, on the torso segment of your player model. Why?